azure-storage-cpp-lite/src/storage_url.cpp

```diff
@@ -2,10 +2,95 @@
 
 namespace microsoft_azure {
     namespace storage {
+        bool is_alnum(char ch)        
```

Comment:
why not to use curl_easy_escape if we are depending on curllib anyway?

Reply:
I just tried out curl_easy_escape, but it it not better than this. You have to initialize curl in every place, and then free the memory with curl_free...plus conversions from std to c_str etc...

Reply:
up to you -) we can always put it into a helper function. but it's really up to you.